tools/scripts/aducm.mk

```diff
@@ -0,0 +1,179 @@
+$(error $(NEWLINE)$(NEWLINE)CCES_HOME not found at c:/Analog Devices/[CrossCore...]\
+		$(NEWLINE)$(NEWLINE)\
+Please run command "set CCES_HOME=c:\Analog Devices\[CrossCore...]"$(NEWLINE)\
+Ex: set CCES_HOME=c:\Analog Devices\[CrossCore...] Embedded Studio 2.8.0$(NEWLINE)$(NEWLINE))
+endif
+#	LINUX
+else
+copy_fun = cp $(1) $(2)
+remove_fun = rm -rf $(1)
+CCES_HOME ?= $(wildcard /opt/analog/cces/*)
+ifeq ($(CCES_HOME),)
+$(error $(NEWLINE)$(NEWLINE)CCES_HOME not found at /opt/analog/cces/[version_number]\
+		$(NEWLINE)$(NEWLINE)\
+		Please run command "export CCES_HOME=[cces_path]"$(NEWLINE)\
+		Ex: export CCES_HOME=/opt/analog/cces/2.9.2$(NEWLINE)$(NEWLINE))
+endif
+endif
+
+#Set PATH variables where used binaries are found
+OPENOCD_SCRIPTS = $(CCES_HOME)/ARM/openocd/share/openocd/scripts
+OPENOCD_BIN = $(CCES_HOME)/ARM/openocd/bin
+CCES_EXE = $(CCES_HOME)/Eclipse
+export PATH := $(CCES_EXE):$(OPENOCD_SCRIPTS):$(OPENOCD_BIN):$(PATH)
+
+#------------------------------------------------------------------------------
+#                           ENVIRONMENT VARIABLES                              
+#------------------------------------------------------------------------------
+PLATFORM	= aducm3029
+PROJECT_NAME	= $(notdir $(CURDIR))
+PROJECT_PATH	= project
+NO-OS		= $(realpath ../..)
+SCRIPTS_DIR	= $(NO-OS)/tools/scripts
+SCRIPTS_PATH	= $(SCRIPTS_DIR)/platform/$(PLATFORM)
+BINARY		= $(PROJECT_PATH)/Release/$(PROJECT_NAME)
+WORKSPACE	= ..
+CREATE_PROJECT_TARGET = $(PROJECT_PATH)/.cproject
+
+PROJECT		= $(NO-OS)/projects/$(PROJECT_NAME)
+DRIVERS		= $(NO-OS)/drivers
+
+define NEWLINE
+
+
+endef
+
+#------------------------------------------------------------------------------
+#                           MAKEFILE SOURCES                              
+#------------------------------------------------------------------------------
+include src.mk
+
+
+INCLUDE_SRC_PATTERN = -append-switch compiler -I=$(src_folder)
+#Include all sourcefile directories
+INCLUDES = $(foreach src_folder, $(SRC_FOLDERS),$(INCLUDE_SRC_PATTERN))
+#Include all include directories added by each makefile
+INCLUDES += $(foreach src_folder, $(INCLUDE_FOLDERS),$(INCLUDE_SRC_PATTERN))
+#Include app_directory
+INCLUDES +=  $(foreach src_folder, $(PROJECT)/$(APP_SRCS),$(INCLUDE_SRC_PATTERN))
+
+LINK_SRC_PATTERN = -link $(src_folder) $(patsubst $(NO-OS)/%,noos/%,$(src_folder))
+#Link all src directories
+SOURCES = $(foreach src_folder, $(SRC_FOLDERS),$(LINK_SRC_PATTERN))
+#link app src_folder
+SOURCES += -link $(PROJECT)/$(APP_SRCS) app_src
+
+#------------------------------------------------------------------------------
+#                           RULES                              
+#------------------------------------------------------------------------------
+
+# Build project Release Configuration
+PHONY := all
+all: update
+	cces -nosplash -application com.analog.crosscore.headlesstools \
+	-data $(WORKSPACE) \
+	-project $(PROJECT_PATH) \
+	-build Release
```

Comment:
I don't see a way to add compiler or linker flags

Reply:
This command just builds the projects.
Flags are added in the update target like:
-append-switch compiler -D=SOMETHING.
for example $(INCLUDES) is a variable with all the -I flags

Reply:
In the new version is more visible